sdk/containerregistry/azure-containerregistry/azure/containerregistry/aio/_async_base_client.py

```diff
@@ -0,0 +1,57 @@
+# coding=utf-8
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+
+from enum import Enum
+
+from ._async_authentication_policy import ContainerRegistryChallengePolicy
+from .._generated.aio import ContainerRegistry
+from .._user_agent import USER_AGENT
+
+
+class ContainerRegistryApiVersion(str, Enum):
+    """Container Registry API version supported by this package"""
+
+    V0_PREVIEW = ""
+
+
+class ContainerRegistryBaseClient(object):
+    """Base class for ContainerRegistryClient and ContainerRepositoryClient
+
+    :param endpoint: Azure Container Registry endpoint
+    :type endpoint: str
+    :param credential: AAD Token for authenticating requests with Azure
+    :type credential: :class:`azure.identity.DefaultTokenCredential`
+
+    """
+
+    def __init__(self, endpoint, credential, **kwargs):  # pylint:disable=client-method-missing-type-annotations
+        auth_policy = ContainerRegistryChallengePolicy(credential, endpoint)
+        self._client = ContainerRegistry(
+            credential=credential,
+            url=endpoint,
+            sdk_moniker=USER_AGENT,
+            authentication_policy=auth_policy,
+            credential_scopes=kwargs.pop("credential_scopes", ["https://management.core.windows.net/.default"]),
```

Comment:
Do you intend `credential_scopes` to be public?

Reply:
No, changing to private.